src/gadgets/Shiki/MediaWiki:Gadget-Shiki.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * MediaWiki Gadget Shiki.js Code Highlighter
+ * @author Dragon-Fish <dragon-fish@qq.com>
+ * @license MIT
+*/
+
+/* eslint-disable prefer-arrow-functions/prefer-arrow-functions */
+/* eslint-disable no-use-before-define */
+
+"use strict";
+
+(async () => {
+    const SHIKI_CDN = "https://esm.sh/shiki@1.24.0";
+    const TARGET_ELEMENTS = document.querySelectorAll(["pre.highlight", "pre.hljs", "pre.prettyprint", "pre.mw-code", "pre[lang]", "code[lang]", "pre[data-lang]", "code[data-lang]"]);
+
+    const shiki = await import(SHIKI_CDN);
+    TARGET_ELEMENTS.forEach((el) => {
+        renderBlock(shiki, el);
+    });
+    mw.hook("npm:shiki").fire(shiki);
+
+    const getLangFromContentModel = () => {
+        const nsNumber = mw.config.get("wgNamespaceNumber");
+        const pageName = mw.config.get("wgPageName");
+        const contentModel = mw.config.get("wgPageContentModel", "").toLowerCase();
+        if (pageName.endsWith(".js") || contentModel === "javascript") {
+            return "javascript";
+        } else if (pageName.endsWith(".css") || contentModel === "css") {
+            return "css";
+        } else if (
+            // Lua
+            (nsNumber === 828 || ["scribunto", "lua"].includes(contentModel))
+            && !pageName.endsWith("/doc")
+        ) {
+            return "lua";
+        }
+    };
+
+    /**
+     * @param {HTMLElement} el
+     */
+    const getLangFromElement = (el) => {
+        const lang = el.getAttribute("lang") || el.dataset.lang || Array.from(el.classList).find((c) => c.startsWith("language-") || c.startsWith("lang-"));
+        if (lang) {
+            return lang.includes("-") ? lang.split("-")[1] : lang;
+        }
+        return "";
+    };
+
+    /**
+     * @param {import('shiki')} shiki
+     * @param {HTMLElement} el
+     * @returns {Promise<HTMLElement | null>}
+     */
+    async function renderBlock(shiki, el) {
+        if (el.classList.contains("shiki") || !!el.dataset.shikiRendered) {
+            return Promise.resolve(null);
+        }
+        const lang = getLangFromElement(el) || getLangFromContentModel();
+        if (!lang) {
+            return Promise.resolve(null);
+        }
+
+        const langInfo = shiki.bundledLanguagesInfo.find((i) => i.aliases?.includes(lang) || i.id === lang || i.name === lang);
+        const langLabel = (() => {
+            if (!langInfo) {
+                return lang;
+            }
+            return [langInfo.aliases?.[0], langInfo.name, langInfo.id, lang].filter(Boolean).sort((a, b) => a.length - b.length)[0];
+        })();
+        console.info("[SHIKI]", "Rendering", el, lang, langInfo);
```

Comment:
能不能不要拉屎（至少改成`console.debug`吧（
```suggestion
        console.debug("[SHIKI]", "Rendering", el, lang, langInfo);
```